packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -83,142 +88,69 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
-        embedded: &Embedded<'s>,
-        name: &str,
-        pkg_type: PackageType,
-    ) -> Result<Self> {
+    fn import(scope: &mut v8::HandleScope<'s>, name: &str, pkg_type: PackageType) -> Result<Self> {
         let path = get_pkg_path(name, pkg_type);
         tracing::debug!("Importing package from path `{path}`");
-        let code = match fs::read_to_string(path.clone()) {
-            Ok(s) => s,
-            Err(_) => {
-                tracing::debug!("Couldn't read package file. It might intentionally not exist.");
-                // Packages don't have to use JS.
-                let fns = v8::Array::new(scope, 3);
-                let undefined = v8::undefined(scope).into();
-                for fn_idx in 0..3 {
-                    fns.set_index(scope, fn_idx, undefined).ok_or_else(|| {
-                        Error::PackageImport(
-                            path.clone(),
-                            format!(
-                                "Could not set Undefined value at index {fn_idx} on package \
-                                 function array"
-                            ),
-                        )
-                    })?;
-                }
-
-                return Ok(JsPackage { fns });
-            }
-        };
-
-        // Avoid JS ReferenceError by wrapping potentially undeclared variables with `typeof`.
-        // Double braces like `{{` are Rust's escape string for a single literal `{`.
-        let wrapped_code = new_js_string(
-            scope,
-            &format!(
-                "((hash_util, hash_stdlib) => {{{code}
-                    return [
-                        typeof start_experiment === 'undefined' ? undefined : start_experiment,
-                        typeof start_sim === 'undefined' ? undefined : start_sim,
-                        typeof run_task === 'undefined' ? undefined : run_task
-                    ]
-                }})",
-            ),
-        );
-
-        let pkg: Function<'_> = {
-            let mut try_catch = v8::TryCatch::new(scope);
-
-            let pkg = v8::Script::compile(&mut try_catch, wrapped_code, None).ok_or_else(|| {
-                let exception = try_catch.exception().unwrap();
-                let exception_string = exception
-                    .to_string(&mut try_catch)
-                    .unwrap()
-                    .to_rust_string_lossy(&mut try_catch);
-
-                Error::PackageImport(
-                    path.clone(),
-                    format!("Could not compile code for package: {exception_string}"),
-                )
-            })?;
 
-            let pkg = pkg.run(&mut try_catch).ok_or_else(|| {
-                let exception = try_catch.exception().unwrap();
-                let exception_string = exception
-                    .to_string(&mut try_catch)
-                    .unwrap()
-                    .to_rust_string_lossy(&mut try_catch);
+        let namespace: Object<'_> = {
```

Comment:
Can we move out this into a function? something like `get_module_namespace`?

Reply:
Added a function https://github.com/hashintel/hash/pull/493/commits/e5aa83ab7819fe8c7a174c52f692c119c4cdd939